core/kernel/dt.c

```diff
@@ -51,6 +51,101 @@ bool dt_have_prop(const void *fdt, int offs, const char *propname)
 	return prop;
 }
 
+int dt_get_irq(void *fdt, int node)
+{
+	const uint32_t *int_prop = NULL;
+	int len_prop = 0;
+	int it_num = -1;
+
+	/*
+	 * Interrupt property is defined with 3x32 bits word:
+	 *  - Type of interrupt
+	 *  - Interrupt Number
+	 *  - Interrupt Trigger type
+	 */
+	int_prop = fdt_getprop(fdt, node, "interrupts", &len_prop);
+
+	if (!int_prop || (len_prop < 2))
+		return -1;
+
+	it_num = fdt32_to_cpu(int_prop[1]);
+
+	return it_num;
+}
+
+paddr_t dt_node_offset_by_compatible_status(void *fdt, int status, int *node,
```

Comment:
This is a bit strange to return value of property `reg` whereas function label should return the node offset. Once could use this function to find a node without such a `reg` property.

Suggest to remove this part from this function and let caller call `_fdt_reg_base_address()` after this function successfully returned.

Reply:
Thanks for pointing this. I prefer to change the function name to dt_reg_base_address_by_compatible_status and rename variable offset by pbase.

Reply:
Well... i think it's a bit too much for such a generic function.
I really think a generic function should be `dt_node_offset_by_compatible_status()` that only takes care about finding target node. 
Once node is found, one shall is free get info from it, i.e. calling `_fdt_fill_device_info()` to get several device info in one shot (status, reg, clock, ....). I think `_fdt_fill_device_info()` could be updated to get 1st interrupt reference if found.

Reply:
ok, I'll change it to return just the node matching and rename function to df_find_node_by_compatible_and_status